tensorflow/core/kernels/rnn/lstm_ops.cc

```diff
@@ -424,6 +424,10 @@ class LSTMBlockCellOp : public OpKernel {
         ctx, cs_prev_tensor->dims() == 2,
         errors::InvalidArgument("cs_prev_tensor must be rank 2 but is rank ",
                                 cs_prev_tensor->dims(), "."));
+    OP_REQUIRES(ctx, 
```

Comment:
This will now return an error if the tensor is empty - is this what we want?  And is this a breaking change if the original CPU kernel previously returned empty tensors?

We should also add a test for this.

Reply:
Ok, yea, I'm actually not sure what we want here either, but `cs_prev_tensor` is the input that is causing the issue. From what I got testing, there are validations that prevent the other inputs from being empty unless this one is empty. I take it instead of returning an error we want to return an empty tensor for GPU as well as CPU?

Reply:
Well, I guess that depends... if there's precedent here for returning an error for all empty tensors, then we can do that - and just hope that nobody was relying on the previous behavior on CPU.  I'm okay with that.

Let's add some tests for it though.

Reply:
Ok, sounds good. I'll add some tests. I did just confirm it fails with the invalid argument on CPU as well so the behavior is at least the same.